plugins/GSdx/Renderers/HW/GSTextureCache.cpp

```diff
@@ -257,20 +258,18 @@ GSTextureCache::Source* GSTextureCache::LookupSource(const GIFRegTEX0& TEX0, con
 		uint32 psm = TEX0.PSM;
 
 		uint32 bw = TEX0.TBW;
-		int tw = 1 << TEX0.TW;
-		int th = 1 << TEX0.TH;
-		uint32 bp_end = psm_s.bn(tw - 1, th - 1, bp, bw);  // Valid only for color formats
 
 		// Arc the Lad finds the wrong surface here when looking for a depth stencil.
 		// Since we're currently not caching depth stencils (check ToDo in CreateSource) we should not look for it here.
 
 		// (Simply not doing this code at all makes a lot of previsouly missing stuff show (but breaks pretty much everything
 		// else.)
 
-		bool texture_inside_rt = ShallSearchTextureInsideRt();
-
-		for(auto t : m_dst[RenderTarget]) {
-			if(t->m_used && t->m_dirty.empty()) {
+		bool found_t = false;
+		for(auto t : m_dst[RenderTarget])
+		{
+			if(t->m_used && t->m_dirty.empty())
```

Comment:
Memo:
`t->m_dirty.empty()` is too strict and also incorrect for tex in rt.
Indiana Jones requires to relax this to correctly display the map.
The validity rectangle of t used for the tex in rt lookup must then take into account the `t->m_dirty` list.
Initially only `C_32` pixel formats can be considered.